packages/client/src/components/app/CodeScanner.svelte

```diff
@@ -0,0 +1,235 @@
+      <StatusLight positive />
+      {value}
+    </div>
+  {/if}
+
+  {#if allowManualEntry && manualMode}
+    <div class="manual-input">
+      <Input
+        bind:value
+        on:change={() => {
+          dispatch("change", value)
+        }}
+      />
+    </div>
+  {/if}
+
+  {#if value}
+    <ActionButton
+      on:click={() => {
+        dispatch("change", "")
+      }}
+      {disabled}
+    >
+      Clear
+    </ActionButton>
+  {:else}
+    <ActionButton
+      icon="Camera"
+      on:click={() => {
+        showReaderModal()
+      }}
+      {disabled}
+    >
+      {scanButtonText}
+    </ActionButton>
+  {/if}
+</div>
+
+<div class="modal-wrap">
+  <Modal bind:this={camModal} on:hide={hideReaderModal}>
+    <ModalContent
+      title={scanButtonText}
+      showConfirmButton={false}
+      showCancelButton={false}
+    >
+      <div id="reader" class="container" bind:this={videoEle}>
+        <div class="camera-placeholder">
+          <Icon size="XXL" name="Camera" />
+          {#if cameraEnabled === false}
+            <div>Your camera is disabled.</div>
+          {/if}
+        </div>
+      </div>
+      {#if cameraEnabled === true}
+        <div class="code-wrap">
+          {#if value}
+            <div class="scanner-value">
+              <StatusLight positive />
+              {value}
+            </div>
+          {:else}
+            <div class="scanner-value">
+              <StatusLight neutral />
+              Searching for code...
+            </div>
+          {/if}
+        </div>
+      {/if}
+
+      <div slot="footer">
+        <div class="footer-buttons">
+          {#if allowManualEntry}
+            <Button
+              group
+              secondary
+              newStyles
+              on:click={() => {
+                manualMode = !manualMode
+                camModal.hide()
+              }}
+            >
+              Enter Manually
+            </Button>
+          {/if}
+
+          <Button
+            group
+            cta
+            on:click={() => {
+              camModal.hide()
+            }}
+          >
+            Confirm
+          </Button>
+        </div>
+      </div>
+    </ModalContent>
+  </Modal>
+</div>
+
+<style>
+  #reader :global(video) {
+    border-radius: 4px;
+    border: var(--border-light-2);
+    overflow: hidden;
+  }
+  .field-display :global(.spectrum-Tags-item) {
+    margin: 0px;
+  }
+  .footer-buttons {
+    display: flex;
+    grid-area: buttonGroup;
+    gap: var(--spectrum-global-dimension-static-size-200);
+  }
+  .scanner-value {
+    display: flex;
+  }
+  .field-display {
+    padding-top: var(
+      --spectrum-fieldlabel-side-m-padding-top,
+      var(--spectrum-global-dimension-size-100)
+    );
+    margin-bottom: var(--spacing-m);
+  }
+  .camera-placeholder {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border-radius: 4px;
+    border: var(--border-light-2);
```

Comment:
This is a hardcoded light border so it looks off in dark themes. Let's use
```css
border: 2px solid var(--spectrum-global-color-gray-300);
```
instead so that it looks better in any theme.